### PR TITLE
[video_player] Add 5.0 interface support

### PR DESCRIPTION
**Pull request by stuartmorgan**
Allows the other video_player packages to use either 4.x or 5.x of
`video_player_platform_interface`, since the only breaking change
doesn't affect them (as it just removes test code that they no longer
use). This allows clients of `video_player` to no longer have a
transitive dependency on test packages, but doesn't create any version
lock with any unendorsed implementations that might exist.

Since this touches `video_player`, it makes the opportunistic change
of cleaning up `CONTRIBUTING.md`, removing discussion of how
to modify federated plugins which is now part of the wiki for the
repository as a whole.

Fixes https://github.com/flutter/flutter/issues/83562

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/video_player/video_player/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 2.2.9
+
+* Adds compatibility with `video_player_platform_interface` 5.0, which does not
```
Comment (cyanglaz) [inline packages/video_player/video_player/CHANGELOG.md:3]: Should we also mentioned the CONTRIBUTING.md change here?

**Review by cyanglaz — APPROVED** [review]
LGTM

File: packages/video_player/video_player/CHANGELOG.md (hunk 1 above)
Comment (stuartmorgan) [reply packages/video_player/video_player/CHANGELOG.md:3]: It seemed extremely niche (the audience would be "people who have contributed to `video_player` in the past, but don't read our current contributing docs"), so I didn't think it was worth mentioning.

**Comment by stuartmorgan** [comment]
Overriding incorrect submit-queue status

**Pull request merged**